core/src/main/java/org/jruby/RubyBoolean.java

```diff
@@ -161,6 +161,19 @@ public static IRubyObject false_xor(IRubyObject f, IRubyObject oth) {
         public static RubyString false_to_s(IRubyObject f) {
             return RubyString.newStringShared(f.getRuntime(), FALSE_BYTES);
         }
+
+        /*
+        Because Boolean objects can't cast to boolean via Class.cast, and you can't return
+        a primitive from a generified method, this impl remains ungenerified.
+         */
+        @Override
+        public Object toJava(Class target) {
+            if (target.isAssignableFrom(Boolean.class) | target.equals(boolean.class)) {
```

Comment:
that one `|` is pbly problematic - also you can skip equals for terminal types and primitives: 
`target == Boolean.class || target == boolean.class`